brmp/pyro_backend.py

```diff
@@ -18,6 +19,18 @@
 from pyro.optim import Adam
 
 
+# Available as `contextmanager.nullcontext` in Python 3.7
+@contextmanager
+def nullcontext(enter_result=None):
+    yield enter_result
+
+
+# A wrapper around the Pyro seed handler, providing handling for the
+# case where seed=None.
+def seed_ctx_mgr(seed):
+    return nullcontext() if seed is None else poutine.handlers.seed(rng_seed=seed)
```

Comment:
We have the [optional](https://github.com/pyro-ppl/numpyro/blob/master/numpyro/util.py#L74) contextmanager in numpyro / pyro for doing this, which you may find useful if this turns out to be a more useful pattern. `with optional(seed is not None, handlers.seed(rng_seed=seed))`

Reply:
Neat, thanks for the pointer.